app/domain/people_skills_filter.rb

```diff
@@ -25,7 +26,33 @@ def filter_by_rated
     entries
   end
 
-  def filter_by_level(entries)
-    level.nil? ? entries : entries.where('level >= ?', level)
+  def filter_by_level(entries, skills)
+    return entries if level.nil?
+    skill_ids = skills.to_s.split(',')
+    if skill_ids.length == @level.length
+      return get_filter_results(entries, skill_ids)
+    else
+      raise ArgumentError, 'Amount of Skill_ids and levels provided do not match'
+    end
+  end
+
+  def get_filter_results(entries, skill_ids)
+    person_ids = []
+    entries.where(buildstring(skill_ids)).group(:person_id).count.each_entry do |key, value|
+      person_ids.push(key) if value == skill_ids.length
+    end
+    entries.where(person_id: person_ids, skill_id: skill_ids)
```

Comment:
How about you do it like so:
```
person_ids = PeopleSkill.where(skill_id: 18, level: 3).or(PeopleSkill.where(skill_id: 12, level: 1)).group_by(&:person_id).transform_values(&:count).map { |id, count| id if count == skill_ids.length }.compact
```
This is obviously crazy for one line. Split the results to several well named variables. But this way you can get all the person_ids you need from one query and don't need some crazy buildstring method.
Of course, you have to do one `or(PeopleSkill.where())` for every array entry.